source/slang/slang-ir-legalize-global-values.cpp

```diff
@@ -0,0 +1,261 @@
+#include "slang-ir-legalize-global-values.h"
+
+#include "slang-ir-clone.h"
+#include "slang-ir-util.h"
+
+namespace Slang
+{
+
+// Returns true iff 'rootInst', or any of it's descendants, satisfies 'predicate'.
+template<typename P>
+bool hasDescendantSatisfyingPredicate(IRInst* rootInst, P predicate)
```

Comment:
I don't think this is necessary. See my comment below.

Reply:
Deleted.

Yeah it shouldn't be needed anymore if my fix to the inlining logic works as I expect.
The fix is detecting if any ancestor of anything used by the inlined value is being inlined, and forces inlining of that thing if so.

That should mean that the inlined values don't use stuff from anywhere under the thing being inlined..